frontend/awx/access/credentials/utils/CredentialsExternalTestModal.tsx

```diff
@@ -0,0 +1,139 @@
+import {
+  IPageAlertToaster,
+  PageFormSelect,
+  PageFormSubmitHandler,
+  PageFormTextArea,
+  PageFormTextInput,
+} from '../../../../../framework';
+import { awxAPI } from '../../../common/api/awx-utils';
+import { usePostRequest } from '../../../../common/crud/usePostRequest';
+import { Credential } from '../../../interfaces/Credential';
+
+export interface CredentialsExternalTestModalProps {
+  credential?: Credential;
+  credentialType: CredentialType;
+}
+
+export interface CredentialsRetainInput {
+  field: {
+    id: number;
+  };
+}
+
+export function CredentialsExternalTestModal(
+  props: CredentialsExternalTestModalProps & { popDialog: () => void } & {
+    alertToaster: IPageAlertToaster;
+  }
+) {
+  const { t } = useTranslation();
+  const postRequest = usePostRequest<CredentialsRetainInput>();
+  const alertToaster = props.alertToaster;
+
+  const onSubmit: PageFormSubmitHandler<CredentialsRetainInput> = async (
+    retainInput: CredentialsRetainInput
+  ) => {
+    const alert: AlertProps = {
+      variant: 'success',
+      title: t('Test passed.'),
+      timeout: 2000,
+    };
+    props.credential
+      ? await postRequest(awxAPI`/credentials/${String(props.credential.id)}/test/`, retainInput)
+          .then(() => {
+            alertToaster.addAlert(alert);
+          })
+          .catch((error) => {
+            alertToaster.addAlert({
+              variant: 'danger',
+              title: t('Something went wrong with the request to test this credential.'),
+              children: error instanceof Error && error.message,
+            });
+          })
+      : await postRequest(
+          awxAPI`/credential_types/${String(props.credentialType.id)}/test/`,
+          retainInput
+        )
+          .then(() => {
+            alertToaster.addAlert(alert);
+          })
+          .catch((error) => {
+            alertToaster.addAlert({
+              variant: 'danger',
+              title: t('Something went wrong with the request to test this credential.'),
+              children: error instanceof Error && error.message,
+            });
+          });
+  };
+  const onCancel = () => props.popDialog();
+
+  return (
+    <Modal
+      aria-label={t(`Test external credential`)}
+      variant="small"
+      position="default"
+      title={t`Test external credential`}
+      hasNoBodyWrapper
+      isOpen
+      onClose={() => props.popDialog()}
+    >
+      <AwxPageForm
+        submitText={t('Run')}
+        onSubmit={onSubmit}
+        cancelText={t('Cancel')}
+        onCancel={onCancel}
+        singleColumn
+      >
+        {props.credentialType.inputs.metadata.map((field: CredentialInputField) => {
+          const isRequired = props.credentialType.inputs?.required.includes(field.id);
+          if (field.type === 'string') {
+            if (field.choices) {
+              return (
+                <PageFormSelect
+                  key={`credential-${field.id}`}
```

Comment:
I believe this was mapped as 
```suggestion
                  key={`${field.id}`}
```
To map react hook with the name of field, or have you change the root of names to add `credential`?